### PR TITLE
chore: jwt rework

### PR DESCRIPTION
**Pull request by alisher-aituarov**
## Summary of change

Updates example apps 

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [ ] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If added a new recipe interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If I added a new recipe, I also added the recipe entry point into the `size-limit` section of `package.json` with the size limit set to the current size rounded up.
-   [ ] If I added a new recipe, I also added the recipe entry point into the `rollup.config.mjs`

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: examples/with-thirdpartyemailpassword-2fa-passwordless/api-server/index.ts [hunk 1]
```diff
@@ -151,6 +151,16 @@ supertokens.init({
...
+                        getSession: async (input) => {
+                            const result = await originalImplementation.getSession(input);
+                            if (result) {
+                                const origPayload = result.getAccessTokenPayload();
+                                if (origPayload.appSub === undefined) {
+                                    await result.mergeIntoAccessTokenPayload({ appSub: origPayload.sub, sub: null });
+                                }
+                            }
+                            return result;
+                        },
```
Comment (porcellus) [inline examples/with-thirdpartyemailpassword-2fa-passwordless/api-server/index.ts:163]: why is this required?

File: examples/with-hasura-thirdpartyemailpassword/src/Home/CallAPIView.js [hunk 2]
```diff
@@ -3,14 +3,21 @@ import Session from "supertokens-auth-react/recipe/session";
...
-
+        let hasuraTokenPayload = await Session.getAccessTokenPayloadSecurely();
+        let jwt = null;
+        if (hasuraTokenPayload.jwt === undefined) {
+            jwt = await Session.getAccessToken();
+        } else {
+            // This branch is only required if there are valid access tokens created before the update
+            // It can be removed after the validity period ends
+            jwt = hasuraTokenPayload.jwt;
+        }
```
Comment (porcellus) [inline examples/with-hasura-thirdpartyemailpassword/src/Home/CallAPIView.js:14]: We don't really need to migration code in our examples.

```suggestion
        const jwt = Session.getAccessToken();
```

File: examples/with-hasura-thirdpartyemailpassword/api-server.js [hunk 3]
```diff
@@ -52,11 +52,18 @@ supertokens.init({
             ],
         }),
         Session.init({
-            jwt: {
-                enable: true,
-                issuer: process.env.JWT_ISSUER_URL,
-            },
+            getTokenTransferMethod: () => "header",
```
Comment (porcellus) [inline examples/with-hasura-thirdpartyemailpassword/api-server.js:55]: Why did this change?

```suggestion
            exposeAccessTokenToFrontendInCookieBasedAuth: true,
```

File: examples/with-hasura-thirdpartyemailpassword/api-server.js [hunk 4]
```diff
@@ -52,11 +52,18 @@ supertokens.init({
...
             override: {
+                openIdFeature: {
+                    functions: (originalImplementation) => ({
+                        ...originalImplementation,
+                        getOpenIdDiscoveryConfiguration: async (input) => ({
+                            issuer: process.env.JWT_ISSUER_URL,
+                            jwks_uri: `${apiDomain}/auth/jwt/jwks.json`,
```
Comment (porcellus) [inline examples/with-hasura-thirdpartyemailpassword/api-server.js:62]:
```suggestion
                            jwks_uri: `${process.env.JWT_ISSUER_URL}/auth/jwt/jwks.json`,
```

File: examples/with-emailverification-then-password-thirdpartyemailpassword/api-server.js [hunk 5]
```diff
@@ -171,6 +171,7 @@ supertokens.init({
             ],
         }),
         Session.init({
+            exposeAccessTokenToFrontendInCookieBasedAuth: true,
```
Comment (porcellus) [inline examples/with-emailverification-then-password-thirdpartyemailpassword/api-server.js:174]: why is this required?

**Review by porcellus — CHANGES_REQUESTED** [review]

File: examples/with-hasura-thirdpartyemailpassword/api-server.js [hunk 6]
```diff
@@ -52,11 +52,18 @@ supertokens.init({
...
-                issuer: process.env.JWT_ISSUER_URL,
-            },
+            exposeAccessTokenToFrontendInCookieBasedAuth: true,
             override: {
+                openIdFeature: {
+                    functions: (originalImplementation) => ({
+                        ...originalImplementation,
+                        getOpenIdDiscoveryConfiguration: async (input) => ({
+                            issuer: process.env.JWT_ISSUER_URL,
+                            jwks_uri: `${process.env.JWT_ISSUER_URL}/auth/jwt/jwks.json`,
```
Comment (porcellus) [inline examples/with-hasura-thirdpartyemailpassword/api-server.js:62]:
```suggestion
                            jwks_uri: `${process.env.JWT_ISSUER_URL}/jwt/jwks.json`,
```

**Review by porcellus — CHANGES_REQUESTED** [review]

**Review by porcellus — APPROVED** [review]

File: examples/with-hasura-thirdpartyemailpassword/src/Home/CallAPIView.js [hunk 7]
```diff
@@ -3,14 +3,13 @@ import Session from "supertokens-auth-react/recipe/session";
 
 export default function CallAPIView() {
     async function callAPIClicked() {
-        let hasuraJWT = (await Session.getAccessTokenPayloadSecurely())["jwt"];
-
+        const jwt = Session.getAccessToken();
```
Comment (porcellus) [inline examples/with-hasura-thirdpartyemailpassword/src/Home/CallAPIView.js:6]:
```suggestion
        const jwt = await Session.getAccessToken();
```

**Review by porcellus — CHANGES_REQUESTED** [review]

**Review by porcellus — APPROVED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**